### ISSUE TITLE
Linux with XDG Desktop Portal: set_directory() only works with save dialog

### ISSUE THREAD

**Issue by sourcebox** [PolyMeilex/rfd#213]
`set_directory()` only works with save dialogs, not with open dialog. I'm not sure if this is really a bug in `rfd` but in XDG portal itself. According the [docs](https://docs.rs/rfd/latest/rfd/#state) setting the starting path should be supported in general, the rustdoc comments still show it as "GTK-only" though.

**Comment by bilelmoussaoui** [comment]
That is not true https://bilelmoussaoui.github.io/ashpd/ashpd/desktop/file_chooser/struct.OpenFileRequest.html#method.current_folder, the option is either not used by rfd or you have an old portal installed that doesn't support it.

**Comment by lukexor** [comment]
See latest on #237
